.github/workflows/aws-deploy.yml

```
@@ -26,15 +26,33 @@ permissions:
     contents: write  # Required for actions/checkout and OIDC tokens
 
 jobs:
+    build:
+      runs-on: ubuntu-latest
+
+      steps:
+        - name: Checkout
+          uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
+  
+        - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8
+          with:
+            node-version: "18"
+  
+        - name: Install and Build
+          run: npm ci && npx vite build
+  
+        - name: Upload build files
+          uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3
+          with:
+            name: aws-deploy-files
+            path: ./dist
+    
     deploy:
         runs-on: ubuntu-latest
```

```suggestion
        runs-on: ubuntu-latest
        needs: build
```

Per [these docs](https://docs.github.com/en/actions/using-workflows/about-workflows#creating-dependent-jobs), since the default behavior is for jobs to run in parallel, then this workflow currently fails because the `deploy` job runs before the `build` job has created the artifact.

---

Just fixed this, thanks for reminding me that I can test the deployment without needing to merge it :X